### ISSUE TITLE
TryIt doesn't send request body for DELETE requests

### ISSUE THREAD

**Issue by iamhosseindhv** [stoplightio/elements#2230]
## Context

When making an api request with method DELETE which has some body, the body is not sent. This seems to be intentional as per these lines:
https://github.com/stoplightio/elements/blob/b20ba4422985ff35617171ec72103719fe3aa2b6/packages/elements-core/src/components/TryIt/build-request.ts#L66

## Current Behavior

As explained above

## Expected Behavior

Request body should be sent for DELETE requests. It's worth noting that DELETE requests with request body is NOT against the specs. There might be many legacy apis that developers are documenting, which may have a DELEET endpoint with request body. Open Api Spec also supports DELETE requests with body.  https://spec.openapis.org/oas/v3.1.0#version-3-1-0

<img width="809" alt="Screenshot 2022-09-05 at 15 22 20" src="https://user-images.githubusercontent.com/26996372/188470506-a978775c-cc9b-4a3d-9795-bd560cd76801.png">

## Steps to Reproduce

1. Use try-it widget to make a DELETE request with request body. Note how in the screenshot below, it is documented that the endpoint accepts some body, but "Request Sample" does not reflect the same

<img width="1339" alt="Screenshot 2022-09-05 at 15 23 51" src="https://user-images.githubusercontent.com/26996372/188470779-2fd8f858-36e5-48d2-9c97-75e7c8e865c8.png">

## Environment

- Version used:
- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile):
- Link to your environment/workspace/project:

**Comment by mnaumanali94** [comment]
We can potentially create a rule in Spectral to warn users but support it in the docs.

**Comment by philsturgeon** [comment]
@mnaumanali94 I agree, thats a good way to split it. When there is a SHOULD that means people could. Spectral can help make things better, and Elements should show whatever it understands regardless of opinion. IMO. :D

**Comment by Invincibear** [comment]
Any updates on this?

**Comment by jpjpjp** [comment]
FWIW, this seems to still be happening (two+ years later).

**Comment by dbergey** [comment]
I ran into this as well with a GET request.

**Comment by cmark1302** [comment]
It's better to show warning and allow the request body.

**Comment by sjayach** [comment]
@mnaumanali94 here is the MR. Please review it when you get a chance 

https://github.com/stoplightio/elements/pull/2785

**Comment by dbergey** [comment]
I made a PR for both GET and DELETE requests here: https://github.com/stoplightio/elements/pull/2777
